internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -1635,7 +1684,9 @@ public void sendControlMessage(
               callback,
               updateCheckSum,
               leaderMetadataWrapper,
-              VENICE_DEFAULT_LOGICAL_TS).get();
+              VENICE_DEFAULT_LOGICAL_TS,
+              isStartOfSegment, // addLeaderCompletedStatus for SOS
```

Comment:
So we want to add this header for every SOS message besides HB messages? What is the reason behind this behavior?

Reply:
Regarding `all control and data messages such that the follower is not waiting for heartbeats and can complete sooner than later.`, it is not completely true since the PR will send a HB message with `completed` header right away once it is ready to serve, and HB frequency is not bad (1 min interval), and I recall the interval to create a new segment is 5 mins, so this reason doesn't hold true from my understanding.

Reply:
with the latest diff, only HB has this header. Thanks for the discussion.